sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs

```diff
@@ -216,38 +216,29 @@ protected override async Task<bool> AcceptCommand (MessageId id, string method,
 
 			case "Runtime.getProperties": {
 					var objId = args? ["objectId"]?.Value<string> ();
-					if (objId.StartsWith ("dotnet:", StringComparison.Ordinal)) {
-						var parts = objId.Split (new char [] { ':' });
-						if (parts.Length < 3)
-							return true;
-						switch (parts[1]) {
-						case "scope": {
-							await GetScopeProperties (id, int.Parse (parts[2]), token);
-							break;
-							}
-						case "object": {
-							await GetDetails (id, MonoCommands.GetObjectProperties (int.Parse (parts[2]), expandValueTypes: false), token);
-							break;
-							}
-						case "array": {
-							await GetArrayDetails (id, objId, parts, token);
-							break;
-							}
-						case "valuetype": {
-							await GetDetailsForValueType (id, objId,
-									get_props_cmd_fn: () => {
-										if (parts.Length < 4)
-											return null;
-
-										var containerObjId = int.Parse (parts[2]);
-										return MonoCommands.GetObjectProperties (containerObjId, expandValueTypes: true);
-									}, token);
-							break;
-							}
-						}
+					if (!objId.StartsWith ("dotnet:", StringComparison.Ordinal))
+						// not for us
+						break;
+
+					var parts = objId.Split (new char [] { ':' });
+					if (parts.Length < 3) {
+						// FIXME: um return an error?
+						return false;
+					}
 
+					var result = await RuntimeGetProperties (id, objId, parts, args, token);
+					SendResponse (id, result, token);
+					return true;
+				}
+
+			case "Runtime.releaseObject": {
+					var objId = args ["objectId"]?.Value<string> ();
+					if (objId?.StartsWith ("dotnet:cfo_res:", StringComparison.Ordinal) == true) {
+						await SendMonoCommand (id, new MonoCommands ($"MONO.mono_wasm_release_object('{objId}')"), token);
```

Comment:
The idea behind the admittedly confusing static MonoCommands was a level of type safety for the arguments, is there a reason you aren't adding the commands there?

Reply:
Oops, I missed that!

Reply:
Easy to fix in a follow up.